appsmithorg/appsmith
# fix: Fix dropdown bug issue

The saas datasource editor page re-renders whenever the name of the datasource is changed causing selected options to be cleared off. This PR fixes that issue.

#### PR fixes following issue(s)
Fixes #22576

#### Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video
>
>
#### Type of change
> Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)


>
>
#### Test Plan
1)  Navigate to create new DS 
2) Rename the DS and click on white space 
3) Select scope to authorize the datasource and then change the name of the datasource and click on white space 
4) Rename the DS and click on white space  and check if the scope of other datasource gets changed 
5) Rename the DS and click on the white space click on the back button to ensure the pop-up is displayed to the user
6) rename the DS from entity explore and observe the scope of the same DS and other DS 
7) Ensure user is able to create new API 
8) Ensure user is able to edit the DS  (Scope , authorise to different Gmail Account)


#### Issues raised during DP testing
> Link issues raised during DP testing for better visibility and tracking (copy link from comments dropped on this PR)
>
>
>
## Checklist:
#### Dev activity


#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
